### ISSUE TITLE
UiAutomator2 fails to scroll down the list to the end and throws NoSuchElement exception

### ISSUE THREAD

**Issue by dmlemeshko** [appium/appium-uiautomator2-server#100]
Appium server: 1.7.1
Appium client: wd.js 1.4.1
Env: MacOS Sierra, Android 5.0 emulator, automationName = UiAutomator2

UiAutomator2 fails to scroll down list to the end, throws exception:
```elementByAndroidUIAutomator("new UiScrollable(new UiSelector().scrollable(true).instance(0)).scrollIntoView(new UiSelector().textStartsWith(\"2016\").instance(0))")] Error response status: 7, , NoSuchElement - An element could not be located on the page using the given search parameters. Selenium error: An element could not be located on the page using the given search parameters```

Screenshot shows were it stuck: http://prntscr.com/hjt3y8

Works perfectly with original UiAutomator (automationName=Appium)

**Comment by binubeee** [comment]
I have come across same issue recently. It just gave up halfway through the scroll.

**Comment by dmlemeshko** [comment]
@binubeee it was fixed with https://github.com/appium/appium-uiautomator2-server/commit/209180d80713a444a1f0807614f95f8a441c0d88
Should work with appium server version `1.7.2`

**Comment by Gocoollnath** [comment]
Yes UIAutomator2 leaves scroll in halfway and throws no such element error

**Comment by Gocoollnath** [comment]
Any updates,guys. @mykola-mokhnach @sravanmedarapu
